### DISCUSSION TITLE
list in ListModelMixin could overwrite the builtins list method of python and causes error

### DISCUSSION THREAD

**Discussion by ghostinthefingers** [encode/django-rest-framework#9159] [Potential Issue] [answered]
list in ListModelMixin could overwrite the builtins list method of python and causes error, isn't it better to change the name of `list` here?

```py
class SomeViewSet(viewsets.ModelViewSet):
    def list(self, request, *args, **kwargs):
        return Response(
            {"message": "List function is not offered in this path"}, 
            status.HTTP_405_METHOD_NOT_ALLOWED
        )
```

then wanna use the builtins list function of python in my code and I get an error, because I already overwrote it:
```py
    def create(...):
        return list('123')
```

**Comment by auvipy** [comment]
@encode/django-rest-framework team, what is your thoughts?

**Reply by kevin-brown** [reply]
We're many years into shadowing this and it hasn't come up as a problem until now. You can pull `list` from the builtins namespace if you find that it's been shadowed.

**Reply by kevin-brown** [reply]
Which is a long way to say I don't think the juice is worth the squeeze here.

**Comment by carltongibson** [comment, marked as answer]
Also, `list` is a bound method so it's accessed as `self.list` not `list`, so there's no shadowing.
